File: ig-codegen/src/main/java/io/github/dizuker/igcodegen/PackageManifest.java
package io.github.dizuker.igcodegen;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;
import java.nio.file.Path;
import java.util.Map;

/**
 * A FHIR package manifest ({@code package.json}, restored by Firely Terminal's {@code fhir
 * restore}). Despite the filename, this is the FHIR package ecosystem's manifest shape, not npm's.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record PackageManifest(String name, String version, Map<String, String> dependencies) {

  public static PackageManifest read(Path packageJsonFile, ObjectMapper objectMapper)
      throws IOException {
    return objectMapper.readValue(packageJsonFile.toFile(), PackageManifest.class);
  }
}

File: ig-codegen-python/src/ig_codegen/package_manifest.py
"""A FHIR package manifest (``package.json``, restored by Firely Terminal's ``fhir restore``).
Despite the filename, this is the FHIR package ecosystem's manifest shape, not npm's.
"""

from __future__ import annotations

import json
from dataclasses import dataclass
from pathlib import Path


@dataclass(frozen=True)
class PackageManifest:
    name: str
    version: str
    dependencies: dict[str, str]

    @staticmethod
    def read(package_json_file: Path) -> PackageManifest:
        data = json.loads(package_json_file.read_text())
        return PackageManifest(
            name=data["name"],
            version=data["version"],
            dependencies=data.get("dependencies", {}),
        )
